libvast/src/system/active_partition.cpp

```diff
@@ -123,12 +123,6 @@ pack(flatbuffers::FlatBufferBuilder& builder,
   }
   auto indexes = builder.CreateVector(indices);
   // Serialize layout.
-#if VAST_ENABLE_ASSERTIONS
-  const auto clt = type{combined_layout};
-  VAST_ASSERT(clt.name().empty() && clt.attributes().empty(),
-              "expecting combined layout not to have metadata for "
-              "serialization as legacy record type");
-#endif
```

Comment:
Note for the reviewer: This check was forgotten to be removed late on in the review of #1888; it can never trigger by design, but that design was only introduced late on in the PR so this check is unnecessary.